modules/common/tls/tls.go

```diff
@@ -0,0 +1,235 @@
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &TLS{
+		Service: service,
+		Ca:      ca,
+	}, nil
+}
+
+// EnsureOrCreateSecret - checks if a secret with the given name exists in the specified namespace
+func EnsureOrCreateSecret(clientset *kubernetes.Clientset, namespace, secretName string, data map[string][]byte) error {
+	_, err := clientset.CoreV1().Secrets(namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Secret not found, create it
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: namespace,
+				},
+				Data: data, // The data to store in the secret - the actual data provided by cert-manager formatted as "tls.crt": []byte("..."),  "tls.key": []byte("..."),
+			}
+			_, err := clientset.CoreV1().Secrets(namespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("Error creating secret %s: %w", secretName, err)
+			}
+		} else {
+			return fmt.Errorf("Error validating secret %s: %w", secretName, err)
+		}
+	}
+	return nil
+}
+
+// CreateVolumeMounts - add volume mount for TLS certificate and CA certificates
+func CreateVolumeMounts(tls *TLS) []corev1.VolumeMount {
+	var volumeMounts []corev1.VolumeMount
+
+	if tls.Service.SecretName != "" {
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      "tls-certs",
+			MountPath: "/etc/tls-certs",
+			ReadOnly:  true,
+			SubPath:   "tls.crt",
+		},
+			corev1.VolumeMount{
+				Name:      "tls-certs",
+				MountPath: "/etc/tls-certs",
+				ReadOnly:  true,
+				SubPath:   "tls.key",
+			})
+	}
+
+	if tls.Ca.CaSecretName != "" {
+		for i := 0; i < MaxNumberOfCAs; i++ {
+			volumeMounts = append(volumeMounts, corev1.VolumeMount{
+				Name:      fmt.Sprintf("ca-cert-%d", i),
+				MountPath: fmt.Sprintf("/etc/ca-certs/%d", i),
+				ReadOnly:  true,
+				SubPath:   fmt.Sprintf("ca-%d.crt", i),
+			})
+		}
+	}
+
+	return volumeMounts
+}
+
+// CreateVolumes - add volume for TLS certificate and CA certificates
+func CreateVolumes(tls *TLS) []corev1.Volume {
+	var volumes []corev1.Volume
+
+	if tls.Service.SecretName != "" {
+		volumes = append(volumes, corev1.Volume{
+			Name: "tls-certs",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: tls.Service.SecretName,
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "tls.crt",
+							Path: "tls.crt",
+						},
+						{
+							Key:  "tls.key",
+							Path: "tls.key",
+						},
+					},
+				},
+			},
+		})
+	}
+
+	if tls.Ca.CaSecretName != "" {
+		for i := 0; i < MaxNumberOfCAs; i++ {
+			volumes = append(volumes, corev1.Volume{
+				Name: fmt.Sprintf("ca-cert-%d", i),
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: tls.Ca.CaSecretName,
+						Items: []corev1.KeyToPath{
+							{
+								Key:  fmt.Sprintf("ca-%d.crt", i),
```

Comment:
This assumes that a secret will be stored in a key with a non-standard name, while CA are usually referenced by key "ca.crt". This is the case for e.g. certs created by cert-manager.
We have to support that naming convention and we should assume the CA would always be referenced by key  "ca.crt"